frontend/public/components/secrets/create-secret.tsx

```diff
@@ -88,7 +91,7 @@ const generateSecret = () => {
 };
 
 // withSecretForm returns SubForm which is a Higher Order Component for all the types of secret forms.
-export const withSecretForm = (SubForm, modal?: boolean) => class SecretFormComponent extends React.Component<BaseEditSecretProps_, BaseEditSecretState_> {
+export const withSecretForm = (SubForm, modal?: boolean, showSubFormOnly=false, redirect=true) => class SecretFormComponent extends React.Component<BaseEditSecretProps_, BaseEditSecretState_> {
```

Comment:
booleans that default to true are confusing. I would make this something like `skipRedirect = false`

Reply:
I agree that I think I'm/we're passing too many params into `withSecretForm`, I'll look to create a new form.

Reply:
Are you suggesting I create a new component outside of `create-secret.tsx`?

Reply:
fixed